### DISCUSSION TITLE
How to make a relay invisible for  Alexa

### DISCUSSION THREAD

**Discussion by Wintespe** [arendst/Tasmota#20683] [Q&A]
I have implemented a heating control system with a virtual relay (Komfort) and a physical relay (Heizung). The software is self-compiled locally.

![image](https://github.com/arendst/Tasmota/assets/121858068/d20b3d46-977e-48c5-b5c5-a6f869511d00)

The device configuration looks like this. 

![image](https://github.com/arendst/Tasmota/assets/121858068/60f18641-fbbc-40fc-9390-0417ed417922)

I had to assign 2 Friedly Names in the settings.

![image](https://github.com/arendst/Tasmota/assets/121858068/d96899bc-c010-400b-93b7-a6e3efa1cd6e)

I have instructed Alexa (Amazon Echo) to search for devices.
She found 2 new devices namely **NodeMcu2** and **NodMcu2_Heater**.

However, I don't want the NodMcu2_Heater relay not visible to Alexa.
Is there an option or setting to hide the second relay and if so, which one?

Thanks for help.

**Comment by sfromis** [comment]
Not using Alexa, but IIRC it will use the "friendly name". Just guessing, but you could try removing that field from the relay you do not want to be found.

**Comment by s-hadinger** [comment]
See here in the docs https://tasmota.github.io/docs/Alexa/#connecting-to-alexa

Relays and lights with friendly names beginning with a dollar sign ($) will cause Tasmota to not "announce" them (i.e., be hidden) to Alexa. If they were previously discovered, they will still continue to work. As there are only four friendly names provided in Tasmota, if FriendlyName4 begins with $, component 4 and all higher numbered discoverable components will not be discovered by Alexa.

**Comment by Wintespe** [comment]
**sfromis** Removing this field results in FRIENDLY__NAME_UNKWON for every device and hardware, horrible.
**s-hadinger**Thx, sounds good, will try it.

**Comment by Wintespe** [comment]
I set the name to **$NodeMcu2_Heater** but Alexa finds it. 
I am running Tasmota v 11.0.3 Does the $ sign relate to a later version?

**Comment by Wintespe** [comment]
Since it does not work with the $ sign as I expected, I searched in the PR's and in the source code of Tasmota. It seems that the hidden feature was only implemented in the HUE-Emúlation.

**xdrv_20_hue.ino**
```
// Check whether this device should be reported to Alexa or considered hidden.
// Any device whose friendly name start with "$" is considered hidden
bool HueActive(uint8_t device) {
  if (device > MAX_FRIENDLYNAMES) { device = MAX_FRIENDLYNAMES; }
  return '$' != *SettingsText(SET_FRIENDLYNAME1 +device -1);
```
}

I found no indication of a hidden feature in the Belkin-Wemo emulation. It may be that I have overlooked it. 
Thanks for the tips anyway.

**Comment by Wintespe** [comment]
I have tested the $ sign feature in the HUE-emulation, where it works as described.

**Comment by Wintespe** [comment]
The $ sign feature to hide a device may need to be added to the wemos emulation for multiple devices.
I will gladly provide a bug report if requested.

**Comment by Jason2866** [comment]
Asking @s-hadinger to add the hide '$' to Belkin Emulation should be enough :-)

**Comment by Wintespe** [comment]
I hope s-hadinger does the implementation. Because the _devices_ named **_FRIENDLY__NAME_UNKWON_** in Alexa are already stupid.
It is not possible - on the part of amazon - to determine where the device comes from. No IP, no MAC addresses... I have discussed this with the Alexa service :-(
